pytripgui/empty_patient_vc/empty_patient_cont.py

```diff
@@ -0,0 +1,446 @@
+        self.view.hu_value.enable_validation(validator)
+
+        validator = QIntValidator()
+        validator.setBottom(1)
+        enable_validation_list(
+            validator,
+            [dim[1]["slice_number"], dim[2]["slice_number"], dim[2]["pixel_number_x"], dim[2]["pixel_number_y"]])
+
+        validator = QRegularExpressionValidator(Regex.FLOAT.value)
+        self.view.slice_offset.enable_validation(validator)
+
+        validator = QRegularExpressionValidator(Regex.FLOAT_POSITIVE.value)
+        enable_validation_list(validator, [
+            dim[0]["slice_distance"], dim[0]["pixel_size"], dim[1]["depth"], dim[2]["slice_distance"],
+            dim[2]["pixel_size"]
+        ])
+
+        validator = MultipleOfRegularExpressionValidator(Regex.FLOAT_POSITIVE.value)
+        validator.set_multiple_of(dim[0]["pixel_size"])
+        enable_validation_list(validator, [
+            dim[0]["width"],
+            dim[0]["height"],
+        ])
+        dim[0]["pixel_size"].emit_on_text_change(lambda: validate_list([dim[0]["width"], dim[0]["height"]]))
+
+        validator = MultipleOfRegularExpressionValidator(Regex.FLOAT_POSITIVE.value)
+        validator.set_multiple_of(dim[0]["slice_distance"])
+        enable_validation_list(validator, [dim[0]["depth"]])
+        dim[0]["slice_distance"].emit_on_text_change(lambda: validate_list([dim[0]["depth"]]))
+
+        validator = PixelSizeValidator(Regex.FLOAT_POSITIVE.value)
+        validator.set_additional_validation(self._validate_pixel_size)
+        enable_validation_list(validator, [dim[1]["width"], dim[1]["height"]])
+
+        validator = PixelSizeValidator(Regex.INT_POSITIVE.value)
+        validator.set_additional_validation(self._validate_pixel_size)
+        enable_validation_list(validator, [dim[1]["pixel_number_x"], dim[1]["pixel_number_y"]])
+
+    def _validate_all(self):
+        if self._validate_general_parameters() and self._validate_tab(self.view.dimensions_tabs.current_index):
+            self._calculate_fields(self.view.dimensions_tabs.current_index)
+            return self._validate_vois() and self._validate_vois_cube()
+        return False
+
+    def _validate_general_parameters(self):
+        return self.view.hu_value.validate() and \
+               self.view.slice_offset.validate()
+
+    def _validate_tabs(self):
+        result = True
+        for index in range(len(self.view.dimensions_fields)):
+            if not self._validate_tab(index):
+                result = False
+        return result
+
+    def _validate_tab(self, index):
+        result = True
+        for field in self.view.dimensions_fields[index].values():
+            if not field.validate():
+                result = False
+        return result
+
+    def _validate_pixel_size(self):
+        dim = self.view.dimensions_fields[1]
+        fields = [dim["width"], dim["pixel_number_x"], dim["height"], dim["pixel_number_y"]]
+        if any(not field.text for field in fields):
```

Comment:
This logical conditions is hard to understand at first glance, add some comments